### PR TITLE
Adding UTs for new vuln cve table insert and clear command

### PR DESCRIPTION
**Pull request by MiguelazoDS**
|Related issue|
|---|
|#7514|

## Description
This PR implements 100% of coverage for the methods in  wdb_agents.c, wdb_parser.c and wdb.c

- wdb_agents_insert_vuln_cve
- wdb_agents_clear_vuln_cve
- wdb_parse_vuln_cve
- wdb_exec_stmt_silent
- wdb_init_stmt_in_cache

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
- [X] Review logs syntax and correct language

- Memory tests for Linux
  - [X] Valgrind (memcheck and descriptor leaks check)

- [X] Added unit tests (for new features)

### CODE REVIEW

File: src/unit_tests/wazuh_db/test_wdb.c [hunk 1]
```diff
@@ -399,6 +403,79 @@ void test_wdb_exec_stmt_error(void **state) {
     cJSON_Delete(result);
 }
 
+/* Tests wdb_exec_stmt_silent */
+
+void test_wdb_exec_stmt_silent_success(void **state) {
```
Comment (DProvinciani) [inline src/unit_tests/wazuh_db/test_wdb.c:408]: The function succeeds either if the result of the call to `wdb_step` is `SQLITE_DONE` as well as `SQLITE_ROW`. I think that we should have two test cases to validate both scenarios. Something like...

```
test_wdb_exec_stmt_silent_success_sqlite_done
test_wdb_exec_stmt_silent_success_sqlite_row
```

File: src/unit_tests/wrappers/wazuh/wazuh_db/wdb_agents_wrappers.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/* Copyright (C) 2015-2020, Wazuh Inc.
```
Comment (DProvinciani) [inline src/unit_tests/wrappers/wazuh/wazuh_db/wdb_agents_wrappers.c:1]: Here should be `2021`.

File: src/unit_tests/wrappers/wazuh/wazuh_db/wdb_agents_wrappers.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* Copyright (C) 2015-2020, Wazuh Inc.
```
Comment (DProvinciani) [inline src/unit_tests/wrappers/wazuh/wazuh_db/wdb_agents_wrappers.h:1]: DITTO

File: src/unit_tests/wazuh_db/test_wdb_agents.c [hunk 4]
```diff
@@ -0,0 +1,134 @@
+
```
Comment (DProvinciani) [inline src/unit_tests/wazuh_db/test_wdb_agents.c:1]: Shouldn't we add the file header with information related to the license?

**Review by DProvinciani — CHANGES_REQUESTED** [review]
Please, apply changes according to the comments.

File: src/unit_tests/wazuh_db/test_wdb_agents.c [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+    os_free(data->wdb->id);
+    os_free(data->wdb->db);
+    os_free(data->wdb);
+    os_free(data);
+    return 0;
+}
+
+/* Tests wdb_agents_insert_vuln_cve */
+
+void test_wdb_agents_insert_vuln_cve_fail(void **state)
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb_agents.c:45]: I think the test should represent the type of failure.
i.e.: test_wdb_agents_insert_vuln_cve_statement_init_fail

File: src/unit_tests/wazuh_db/test_wdb_agents.c [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+    will_return(__wrap_wdb_exec_stmt_silent, OS_SUCCESS);
+
+    ret = wdb_agents_insert_vuln_cve(data->wdb, name, version, architecture, cve);
+
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+/* Tests wdb_agents_clear_vuln_cve */
+
+void test_wdb_agents_clear_vuln_cve_fail(void **state)
```
Comment (palaciosjeremias) [inline src/unit_tests/wazuh_db/test_wdb_agents.c:93]: I think the test should represent the type of failure.
i.e.: test_wdb_agents_clear_vuln_cve_statement_init_fail

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Review by DProvinciani — APPROVED** [review]
Looks good!

**Pull request merged**